sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs

```diff
@@ -31,6 +31,12 @@ public enum ServiceVersion
             V2019_05_06_Preview = 1,
         }
     }
+    public partial class SearchConditionalOptions : Azure.Search.Documents.SearchRequestOptions
+    {
+        public SearchConditionalOptions() { }
+        public Azure.ETag? IfMatch { get { throw null; } set { } }
+        public Azure.ETag? IfNoneMatch { get { throw null; } set { } }
```

Comment:
This isn't quite what I was expecting.  I thought we'd have a property like `public Azure.MatchConditions Conditions { get; set; }` that exposed the Azure.Core version.

Reply:
You did say "like a union", but I also didn't see value in another type in the hierarchy just to set these. That said, please see the other thread on not even defining these. I have a draft PR that modifies this based on suggestions elsewhere, doing something similar to AppConfiguration with having just a `bool onlyIfUnchanged`. For `CreateOrUpdate` this works well, but `Delete` is more interesting.